apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -329,7 +330,7 @@ func (m *ModelManagerAPI) createModelNew(
 	}
 
 	// Create the model in the controller database.
-	modelUUID, finaliser, err := m.modelService.CreateModel(ctx, creationArgs)
```

Comment:
I may have misunderstood this, but why is the modelUUID removed from the return here and made required in the model args?

Right now we are creating the model in state and the UUID is generated there, so we don't want to regenerate the UUID in domain, so the UUID was passed (as an optional argument) in ModelCreationArgs and the new model in domain was created with the same ID.

Once we remove model creation in state we will want to create the model UUID every time in domain, so we will want it to be returned from CreateModel and we won't want to pass it in the model args (except during migrations).

Reply:
All of the callers not in test, actually create or pass (in the case of import) in a UUID. Only the tests expect the UUID to be created by domain. Having two potential sources of truth for the UUID is bad.

I had problems when using the wrong one in a txn. The tests did one thing and the real world did another. The UUID in this case is set at line 265. We can chat if you have more questions.